### PR TITLE
Pull the static file server into the CRA proxy as well

### PR DESCRIPTION
**Pull request by JulianKniephoff**
With this, the outer npm-project is finally obsolete. See the updated `README` for how to use it and the proxy server. I abstained from restructuring the project any further to keep the PR focused, but a PR that will do nothing but move things around is in the pipeline.

Note, this is based on #394.

### CODE REVIEW

File: package.json
Comment (lkiesow) [inline package.json:1]: Removing this, we need to remove this from the CI as well:

https://github.com/opencast/opencast-admin-interface/blob/34dd129025a87002328e853e9b77b599fc7bc5d8/.github/workflows/test.yml#L20-L22

File: app/src/setupProxy.js [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+        if (req.method !== "GET") {
+            req.originalMethod = req.method;
+            req.method = "GET";
+            setTimeout(testFiles, 1000, req, res, next);
+        } else {
+            testFiles(req, res, next);
+        }
+    };
+
+const testFiles = express.static(path.join(__dirname, "../../test"));
```
Comment (lkiesow) [inline app/src/setupProxy.js:40]: The path is missing the `/app`:
```suggestion
const testFiles = express.static(path.join(__dirname, "../../test/app"));
```

Without this change, running `npm start` will immediately end in an error.

File: package.json
Comment (JulianKniephoff) [reply package.json:1]: To be fair, this wasn't there when I first created this. ;D

File: app/src/setupProxy.js [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+        target: process.env.PROXY_TARGET || "http://localhost:8080",
+        changeOrigin: true,
+        auth: process.env.PROXY_AUTH || "admin:opencast",
+    })
+    : (req, res, next) => {
+        if (req.method === "POST") {
+            res.status(201);
+        }
+
+        req.url = `/${req.method}/${req.originalUrl}`;
```
[the remark this answers is not in the document]
Comment (JulianKniephoff) [reply app/src/setupProxy.js]: This was probably caused by trying and reverting lots of different things. Fixed now.

File: app/src/setupProxy.js (hunk 1 above)
Comment (JulianKniephoff) [reply app/src/setupProxy.js]: And this was probably missed when separating this from #399.

**Review by lkiesow — APPROVED** [review]

**Pull request merged**